cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/SourcePointClient.java

```diff
@@ -30,18 +32,32 @@
 
     SourcePointClientConfig config;
 
+    ConnectivityManager manager;
+
     private String getRequestUUID(){
         if(!requestUUID.isEmpty()) return requestUUID;
         requestUUID =  UUID.randomUUID().toString();
         return requestUUID;
     }
 
-    SourcePointClient(OkHttpClient httpClient, SourcePointClientConfig config) {
+    SourcePointClient(OkHttpClient httpClient, SourcePointClientConfig config, ConnectivityManager manager) {
         this.httpClient = httpClient;
         this.config = config;
+        this.manager = manager;
+    }
+
+    private boolean hasLostInternetConnection() {
+        if (this.manager == null) {
```

Comment:
is this ever gonna be null here if injected from the builder?

Reply:
@leodoin  I believe in a rare scenario when context is null due to app inactivity(when application is in background for longer time or device screen is locked  and meanwhile device internet is gone) it may be null, it is always better practice to have null check to avoid null pointer.exception.